### ISSUE TITLE
Can't save sketch (sketch corrupted)

### ISSUE THREAD

**Issue by kjhollen** [processing/p5.js-web-editor#428]
A student had a sketch in class today with some errors, which we fixed, but it would not run. It seems that maybe the root cause is that the code is not being saved on the server end, so it's running a stale version of the code instead of what's in the editor.

Original sketch link:
https://alpha.editor.p5js.org/lsm0821/sketches/r19_DZGY-

When I duplicated the sketch wouldn't save there either:
https://alpha.editor.p5js.org/kjhollen/sketches/Bk0I5t4FZ

The console complains of 'Unexpected Token ILLEGAL' error on line 8, even when it is fixed in the editor (see this sketch for repaired code http://alpha.editor.p5js.org/kjhollen/sketches/SySjoKEFW )—so I think the root of the problem is that the sketch doesn't save.

I'm not sure what the repro steps for this are; I'll update this post if I see this come up again.

**Comment by catarak** [comment]
i tried to reproduce this and i wasn't able to. when i fixed the errors (the period instead of the comma and `mousex` instead of `mouseX`, the sketch ran fine. i also found this to be true when i duplicated the sketch. given #419 i think there's something weird happening with duplicating projects, so maybe it's related to that as well.

**Comment by shiffman** [comment]
I just want to note that this error came up in class at ITP this week. Here's a screenshot of the console as @kjhollen reported.

![screen shot 2017-09-13 at 4 26 41 pm](https://user-images.githubusercontent.com/191758/30456917-84d9faec-9973-11e7-9d34-b745007c6cce.png)

I did some testing and only ran into this issue when "auto-refresh" was enabled. I'm hoping to pinpoint it more specifically.

Once this error happens I cannot save any new changes to the sketch or run it, however, I can refresh the page and continue from where it last saved.

**Comment by itp-room15** [comment]
Hello! This is Dan from afternoon ICM. We are trying to determine what causes this and MH had a suggestion that a single quote `'` or underline `_` at the end of a line of code with auto-refresh enabled breaks the project. It can no longer run or be saved.

```javascript
function setup() { 
  createCanvas(400, 400);
  
} 

function draw() { 
  background(220);
  line(100, 100, 200, 200)'
}
```

**Comment by boazsender** [comment]
It looks like this error happens on the client when a sketch containing a JavaScript syntax error gets passed into decomment, a package in the editor that strips comments from the sketch. I proposed a potential short term fix in pull request #456.

**Comment by catarak** [comment]
woohoo it's awesome you figured out the root cause of this!
